src/ui/components/AppWrapper/AppWrapper.tsx

```diff
@@ -451,6 +453,14 @@ const AppWrapper = (props: { children: ReactNode }) => {
     if (keriaConnectUrlRecord) {
       try {
         await Agent.agent.start(keriaConnectUrlRecord.content.url as string);
+
+        const pendingDeletions =
+          await Agent.agent.connections.getConnectionsPendingDeletion();
+
+        for (const id of pendingDeletions) {
+          await Agent.agent.connections.deleteConnectionById(id);
+          dispatch(removeConnectionCache(id));
+        }
```

Comment:
This is still not what I meant. If KERIA is down when we initially connect, we will hit the catch clause and go into the reconnect loop. Once we connect, we won't do this sync.

You need to move this into the `isOnline` useEffect so that once we come online, we check for any outstanding connections pending deletion.